cypress/e2e/smoke/balances.cy.js

```diff
@@ -0,0 +1,228 @@
+          cy.get('td').eq(FIAT_AMOUNT_COLUMN).contains(fiatRegex)
+        })
+
+      // Strict match because other tokens contain "Ether" in their name
+      cy.contains(/^Ether$/)
+        .parents('tr')
+        .within(() => {
+          cy.get('td').eq(TOKEN_AMOUNT_COLUMN).contains('0.3 ETH')
+          cy.get('td').eq(FIAT_AMOUNT_COLUMN).contains(fiatRegex)
+        })
+
+      cy.contains('Uniswap')
+        .parents('tr')
+        .within(() => {
+          cy.get('td').eq(TOKEN_AMOUNT_COLUMN).contains('0.01828 UNI')
+          cy.get('td').eq(FIAT_AMOUNT_COLUMN).contains(fiatRegex)
+        })
+
+      cy.contains('USD Coin')
+        .parents('tr')
+        .within(() => {
+          cy.get('td').eq(TOKEN_AMOUNT_COLUMN).contains('13,636,504 USDC')
+          cy.get('td').eq(FIAT_AMOUNT_COLUMN).contains(fiatRegex)
+        })
+
+      cy.contains('Gnosis')
+        .parents('tr')
+        .within(() => {
+          cy.get('td').eq(TOKEN_AMOUNT_COLUMN).contains('< 0.00001 GNO')
+          cy.get('td').eq(FIAT_AMOUNT_COLUMN).contains(fiatRegex)
+        })
+
+      cy.contains(/^0x$/)
+        .parents('tr')
+        .within(() => {
+          cy.get('td').eq(TOKEN_AMOUNT_COLUMN).contains('1.003 ZRX')
+          cy.get('td').eq(FIAT_AMOUNT_COLUMN).contains(fiatRegex)
+        })
+    })
+  })
+
+  describe('fiat currency can be changed', () => {
+    it('should have ETH as default currency', () => {
+      // First row Fiat balance should not contain EUR
+      cy.get(balanceSingleRow).first().find('td').eq(FIAT_AMOUNT_COLUMN).should('not.contain', '€')
+      // First row Fiat balance should contain DAI
+      cy.get(balanceSingleRow).first().find('td').eq(TOKEN_AMOUNT_COLUMN).contains('DAI')
+    })
+
+    it('should allow changing the currency to EUR', () => {
+      // Click on balances currency dropdown
+      cy.get('[id="currency"]').click()
+
+      // Select EUR
+      cy.get('ul[role="listbox"]').findByText('EUR').click()
+
+      // First row Fiat balance should not contain USDC
+      cy.get(balanceSingleRow).first().find('td').eq(FIAT_AMOUNT_COLUMN).should('not.contain', 'USDC')
+      // First row Fiat balance should contain EUR
+      cy.get(balanceSingleRow).first().find('td').eq(FIAT_AMOUNT_COLUMN).contains('€')
+    })
+  })
+
+  describe('pagination should work', () => {
+    before(() => {
+      // Open the Safe used for testing pagination
+      cy.visit(`/${PAGINATION_TEST_SAFE}/balances`, { failOnStatusCode: false })
+      cy.contains('button', 'Accept selection').click()
+
+      //Ensure the table is fully loaded
+      cy.contains('Rows per page:', { timeout: 10000 })
+    })
+    it('should should allow 25 rows per page and navigate to next and previous page', () => {
+      // Click on the pagination select dropdown
+      cy.get('[aria-haspopup="listbox"]').contains('25').click()
+
+      // Select 25 rows per page
+      cy.get('[aria-haspopup="listbox"]').get('li[role="option"]').contains('25').click()
```

Comment:
Why clicking on the already selected value?

Reply:
True, no need to click the selector. Now is just there to check that the element exists before continuing

Reply:
I added a test for clicking a different value (10) to see if the pagination worked and the table adjusted accordingly